### PR TITLE
Add architecture selection for CPU/NVIDIA in installer

### PR DESCRIPTION
**Pull request by ItsSimko**
Contains: #83 

## Summary by Sourcery

Introduce architecture selection in the installer to support both CPU and NVIDIA installations, and update the CI workflow to handle the creation and packaging of both CPU-only and CUDA-enabled executables.

New Features:
- Add architecture selection for CPU and NVIDIA in the installer, allowing users to choose between CPU and NVIDIA installations.

Enhancements:
- Refactor the installer script to include logic for handling CPU and NVIDIA installation options.

CI:
- Update CI workflow to build and package both CPU-only and CUDA-enabled executables, and upload them as separate assets in the release.

### CODE REVIEW

File: scripts/install.nsi [hunk 1]
```diff
@@ -95,6 +148,7 @@ SectionEnd
 
 ; Define the installer pages
 !insertmacro MUI_PAGE_LICENSE ".\assets\License.txt"
+Page Custom ARCHITECHTURE_SELECT
 !insertmacro MUI_PAGE_DIRECTORY
 !insertmacro MUI_PAGE_INSTFILES
 !define MUI_FINISHPAGE_RUN "$INSTDIR\freescribe-client.exe"
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply scripts/install.nsi:156]: https://github.com/ClinicianFOCUS/FreeScribe/pull/89/commits/0c92ad0315d3d71c8a4036e38e6ef63bcd62e155

Fixed

File: scripts/install.nsi [hunk 2]
```diff
@@ -19,6 +20,50 @@ VIAddVersionKey "FileDescription" "FreeScribe Installer"
...
+
+    ; Register event handlers for radio buttons
+    ${NSD_OnClick} $CPU_RADIO OnRadioClick
+    ${NSD_OnClick} $NVIDIA_RADIO OnRadioClick
+
+    nsDialogs::Show
+FunctionEnd
+
+; Callback function for radio button clicks
+Function OnRadioClick
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply scripts/install.nsi:59]: Works fine

**Comment by ItsSimko** [comment]
@sourcery-ai review

File: src/FreeScribe.client/Model.py [hunk 3]
```diff
@@ -114,12 +114,12 @@ def generate_response(
 
             # reset the model tokens
             self.model.reset()
-
             return response["choices"][0]["message"]["content"]
             
         except Exception as e:
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply src/FreeScribe.client/Model.py:119]: Okay for now.

**Review by yingbull — APPROVED** [review]

**Pull request merged**